extensions/ql-vscode/src/databases/database-fetcher.ts

```diff
@@ -429,6 +459,11 @@ async function getStorageFolder(storagePath: string, urlStr: string) {
     counter++;
     folderName = join(realpath, `${lastName}-${counter}`);
     if (counter > 100) {
```

Comment:
Is there any need to keep the behaviour of calling databases `java-{1-100}` before we switch to another method? This is only for generating new databases, so it doesn't need to be compatible with existing databases, does it?

Why not use `java-${nanoid()}` all the time?

Reply:
The reason for not doing that is that this name may be shown to the user (as mentioned in [the comment](https://github.com/github/vscode-codeql/pull/3456/files#diff-c165a3c0ec497920fff4a40c198416d0c49572366578137ea82e19620767c256R441-R443): "we need to generate a folder name for the unzipped archive, this needs to be human readable since we may use this name as the initial name for the database"). `java-100` is still more readable than `java-odkDB1k61hq7PDJnv0qGY` and would probably make it easier for users to recognize which database they have just added.

Reply:
Thanks for explaining. It seems we use the filename only if there wasn't a name override (which comes when downloading from github) or and the database itself doesn't define a name (I don't know how common this is, but I expect any database built from a git repo will have a name).

Reply:
Another alternative thought I had was what's the relative performance of checking a single file exists vs listing all files in the directory? If we just list all files and then do our checks in memory I would expect it to work fine up to thousands of databases.

Reply:
Yes, that would make sense. I'll change it to `readdir` once and change the maximum counter value to 10,000.